src/memory/memoryManager.ts

```diff
@@ -54,6 +56,33 @@ export class MemorySegmentManager {
     return address.add(dataLen);
   }
 
+  // Insert a value in the memory at the given address and increase
+  // the segment size by 1.
+  insert(address: Relocatable, value: MaybeRelocatable): void {
```

Comment:
Moved insert here and completely removed `insert` from Memory. This is done in order to be able to keep track of the segment length, which wasn't possible when inserting data via the Memory.